packages/components/src/range-control/styles/range-control-styles.ts

```diff
@@ -26,15 +28,25 @@ const rangeHeight = () =>
 	css( { height: rangeHeightValue, minHeight: rangeHeightValue } );
 const thumbSize = 12;
 
-export const Root = styled.div`
+const deprecatedDefaultHeight = ( {
+	__next40pxDefaultSize,
+}: Pick< RangeControlProps, '__next40pxDefaultSize' > ) => {
+	return ! __next40pxDefaultSize && css( { minHeight: rangeHeightValue } );
+};
+
+type RootProps = Pick< RangeControlProps, '__next40pxDefaultSize' >;
+export const Root = styled.div< RootProps >`
 	-webkit-tap-highlight-color: transparent;
-	align-items: flex-start;
+	align-items: center;
 	display: flex;
 	justify-content: flex-start;
 	padding: 0;
 	position: relative;
 	touch-action: none;
 	width: 100%;
+	min-height: 40px;
+
+	${ deprecatedDefaultHeight }
```

Comment:
Are these `min-height` styles necessary? I guess if the root wrapper's height would be set correctly anyway, since it's "pushed" by the height of the number input.

Reply:
This was my solution to @mirka's [point](https://github.com/WordPress/gutenberg/pull/49105#discussion_r1138612084) that the component needs to support the `withInputField={ false }` case. When we have an input you're right, the wrapper's height will be pushed to the correct height automatically. When we render without an input though, we need to somehow enforce that height on the wrapper directly.